src/arrayMethodSort.js

```diff
@@ -4,9 +4,17 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (callback = (a, b) => String(a).localeCompare(String(b))) {
     // write code here
-  };
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i + 1; j < this.length; j++) {
+        if (callback(this[i], this[j]) > 0) {
+          [this[i], this[j]] = [this[j], this[i]]
+        }
+      }
+    }
+
+    return this;
```

Comment:
The current implementation uses a bubble sort algorithm, which is known for its simplicity but is not the most efficient for larger arrays. Consider researching more efficient sorting algorithms like quicksort or mergesort for better performance on larger datasets.